src/logger/__tests__/MetricsLogger.test.ts

```diff
@@ -297,6 +308,141 @@ test('context is preserved across flush() calls', async () => {
   }
 });
 
+test('putMetricWithDimensions metric only', async () => {
+  // arrange
+  const logger = createLoggerWithDefaultDimensions();
+
+  // act
+  logger.putMetricWithDimensions({
+    metrics: { "MyMetric": 100 }
+  });
+
+  await logger.flush();
+
+  // assert
+  expect(sink.events).toHaveLength(1);
+  const evt = sink.events[0];
+  expect(evt.metrics.size).toBe(1);
+  expect(evt.metrics.get("MyMetric")).toBe(100);
+  // everything else should be defaults
+  expect(evt.namespace).toBe(Constants.DEFAULT_NAMESPACE);
+  expect(evt.getDimensions()[0]).toBe(DEFAULT_DIMENSIONS);
+});
+
+test('putMetricWithDimensions single metric with namespace', async () => {
+  // arrange
+  const logger = createLoggerWithDefaultDimensions();
+
+  // act
+  logger.putMetricWithDimensions({
+    metrics: { "MyMetric": 100 },
+    namespace: "My-Namespace"
+  });
+
+  // act
+  await logger.flush();
+
+  // assert
+  expect(sink.events).toHaveLength(1);
+  const evt = sink.events[0];
+  expect(evt.metrics.size).toBe(1);
+  expect(evt.metrics.get("MyMetric")).toBe(100);
+  expect(evt.namespace).toBe("My-Namespace");
+  expect(evt.getDimensions()[0]).toBe(DEFAULT_DIMENSIONS);
+});
+
+
+test('putMetricWithDimensions with single dimensions and default namespace', async () => {
+  // arrange
+  const logger = createLoggerWithDefaultDimensions();
+  const client = 'client';
+
+  // act
+  logger.putMetricWithDimensions({
+    metrics: { Metric1: 100 },
+    dimensions: [{ client }]
+  });
+
+  await logger.flush();
+
+  // assert
+  expect(sink.events).toHaveLength(1);
+  const evt = sink.events[0];
+  expect(evt.metrics.size).toBe(1);
+  expect(evt.metrics.get("MyMetric")).toBe(100);
+  expect(evt.namespace).toBe(Constants.DEFAULT_NAMESPACE);
+  expect(evt.getDimensions()).toBe([{ ...DEFAULT_DIMENSIONS, client }]);
+});
+
+test('putMetricWithDimensions along multiple dimensions', async () => {
+  // arrange
+  const logger = createLoggerWithDefaultDimensions();
+  const client = 'client';
+  const pageType = 'pageType';
+
+  // act
+  logger.putMetricWithDimensions({
+    metrics: {
+      Metric1: 100,
+    },
+    namespace: "My Namespace",
+    dimensions: [
+      { client },
+      { pageType },
+      { client, pageType },
+    ]
+  });
+
+  await logger.flush();
+
+  // assert
+  expect(sink.events).toHaveLength(1);
+  const evt = sink.events[0];
+  expect(evt.metrics.size).toBe(1);
+  expect(evt.metrics.get("MyMetric")).toBe(100);
+  expect(evt.namespace).toBe("My-Namespace");
+  expect(evt.getDimensions()[0]).toBe([
+    { ...DEFAULT_DIMENSIONS, client },
+    { ...DEFAULT_DIMENSIONS, pageType },
+    { ...DEFAULT_DIMENSIONS, client, pageType },
```

Comment:
This is the part I'm not certain of. One question I have is what are users most likely to do? Do users prefer the default dimensions are included most of the time or are users overwriting them more often than not.

Reply:
This is a tough one 😄

I don't use the default dimensions and having `true` by default will make me very happy. Nonetheless, this is a personal preference and I guess we should reason in terms of user expectations.

One way to see it could be to keep the same behaviour that `putMetric` and `putDimensions` have perhaps? Which means, the default should be `false`. Also, given the wording of the property is `stripDefaultDimensions` it entails that by default it disabled and if you want to activate it you must specify true. The opposite would have been `keepDefaultDimensions` with default set to false, and activation to true.

For that don't use the default dimensions, I would expect to either set `stripDefaultDimensions` explicitly to `true` every time I call `putMetricWithDimensions`, or call `setDimensions()` once at the beginning and then regardless of what the value of `stripDefaultDimensions` is, the default shouldn't be set because they have been removed for the whole session.

Reply:
@jaredcnance any thoughts on this?

Reply:
Right now I’m leaning towards the proposal as is. I think it _may_ make sense to offer a global configuration to disable default dimensions entirely. I suspect you either want them or you don’t and you may want that applied uniformly.